src/mscorlib/src/System/Runtime/InteropServices/NativeLibrary.cs

```diff
@@ -0,0 +1,365 @@
+
+            charSet = default;
+            callingConvention = default;
+
+            var attr = delegateType.GetCustomAttribute<UnmanagedFunctionPointerAttribute>(inherit: false);
+            if (attr != null)
+            {
+                charSet = attr.CharSet;
+                callingConvention = attr.CallingConvention;
+            }
+
+            // If no character set or calling convention was specified (or if the delegate explicitly asked
+            // to use the OS default calling convention), fall back to the OS defaults. We need to check
+            // for default because the developer could have passed '0' for any enum value, and during marshaling
+            // the CLR treats this as "use defaults".
+
+            if (charSet == default)
+            {
+                charSet = FallbackCharSet;
+            }
+
+            if (callingConvention == default || callingConvention == CallingConvention.Winapi)
+            {
+                callingConvention = FallbackCallingConvention;
+            }
+        }
+
+        /// <summary>
+        /// Equivalent to <see cref="TryGetDelegate{TDelegate}(string, bool, out TDelegate)"/> where <em>exactSpelling</em> is <see langword="false"/>.
+        /// </summary>
+        public bool TryGetDelegate<TDelegate>(string name, out TDelegate result) where TDelegate : class
+            => TryGetDelegate<TDelegate>(name, exactSpelling: false, out result);
+
+        /// <summary>
+        /// Attempts to locate a named function in the module's export table and return a managed delegate to that function.
+        /// </summary>
+        /// <typeparam name="TDelegate">The delegate type that will be created. This type must subclass <see cref="MulticastDelegate"/>.
+        /// This type <em>should</em> be (but is not required to be) annotated with <see cref="UnmanagedFunctionPointerAttribute"/>.</typeparam>
+        /// <param name="name">The name of the function to look up.</param>
+        /// <param name="exactSpelling"><see langword="false"/> (default) if the runtime should automatically perform name decoration when
+        /// locating the exported function; <see langword="true"/> if the runtime should use the provided name as-is with no additional
+        /// decoration. See also <see cref="DllImportAttribute.ExactSpelling"/>.</param>
+        /// <param name="result">A <typeparamref name="TDelegate"/> that represents the exported function,
+        /// or <see langword="null"/> if the function cannot be located in the export table.</param>
+        /// <returns><see langword="true"/> if the function was found, otherwise <see langword="false"/>.</returns>
+        /// <remarks>
+        /// Functions can also be queried by ordinal export by using a "#" prefix (e.g., "#1") for <paramref name="name"/>.
+        /// </remarks>
+        public bool TryGetDelegate<TDelegate>(string name, bool exactSpelling, out TDelegate result) where TDelegate : class
```

Comment:
this is a little concern for me. since it is very easy for customer to forgot to add CallConvention attribute for this Delegate Type. Maybe we should add a MSDN page to show how to use this API correctly.

Reply:
The `Marshal.GetDelegateForFunctionPointer` API has a fallback behavior if the `[UnmanagedFunctionPointer]` attribute isn't present. This API tries to honor that same logic to the best of its ability. I agree that documentation will be necessary for correct usage of this API.